File: front/src/api/llmConfig.ts
import request from '@/utils/request'
import type { ApiResponse } from '@/types/api'

export interface LlmConfig {
  id: number
  userId?: number
  name: string
  provider: string
  apiKey: string
  modelName: string
  baseUrl?: string
  isDefault: boolean
  multimodal: boolean
  secretKey?: string
  isSystem: boolean
  createdAt: string
  updatedAt: string
}

export function getLlmConfigs() {
  return request.get<unknown, ApiResponse<LlmConfig[]>>('/llm-configs')
}

export function getLlmConfig(id: number) {
  return request.get<unknown, ApiResponse<LlmConfig>>(`/llm-configs/${id}`)
}

export function createLlmConfig(data: Partial<LlmConfig>) {
  return request.post<unknown, ApiResponse<LlmConfig>>('/llm-configs', data)
}

export function updateLlmConfig(id: number, data: Partial<LlmConfig>) {
  return request.put<unknown, ApiResponse<LlmConfig>>(`/llm-configs/${id}`, data)
}

export function deleteLlmConfig(id: number) {
  return request.delete<unknown, ApiResponse<void>>(`/llm-configs/${id}`)
}

File: back/src/main/java/com/hznan/mamgareader/model/entity/LlmConfig.java
package com.hznan.mamgareader.model.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.*;

import java.time.LocalDateTime;

@TableName("llm_configs")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class LlmConfig {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long userId;

    private String name;

    private String provider;

    private String apiKey;

    private String modelName;

    private String baseUrl;

    @Builder.Default
    private Boolean isDefault = false;

    @Builder.Default
    private Boolean multimodal = false;

    private String secretKey;

    @Builder.Default
    private Boolean isSystem = false;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}
